service/history/api/queryworkflow/api.go

```diff
@@ -149,6 +150,8 @@ func Invoke(
 		return nil, serviceerror.NewWorkflowNotReady("Unable to query workflow due to Workflow Task in failed state.")
 	}
 
+	priority := mutableState.GetExecutionInfo().Priority
```

Comment:
In some cases we've had problems with stuff like this, which just keeps a pointer to the proto (shared with the mutable state cache) instead of cloning it. As long as we treat Priority messages as immutable and always construct new ones, this is safe.